cli/internal/codegen/ts.go

```diff
@@ -506,11 +538,11 @@ func (ts *ts) writeTyp(ns string, typ *schema.Type, numIndents int) {
 			if n := field.Typ.GetNamed(); n != nil {
 				recursiveType = ts.typs.IsRecursiveRef(ts.currDecl.Id, n.Id)
 			}
-			if field.Optional || recursiveType {
-				ts.WriteString("?")
-			}
 			ts.WriteString(": ")
 			ts.writeTyp(ns, field.Typ, numIndents+1)
+			if field.Optional || recursiveType {
+				ts.WriteString(" | undefined")
+			}
```

Comment:
This swap from `?` to ` | undefined` is to bring it more in line for strict typescript.

```typescript
type Old = interface {
   fieldName string
   typeName? string
}

type New = interface {
   fieldName string
   typeName  string | undefined
}

let a: Old = {
  fieldName: "test",
  typoName: "error", // this is allowed 
}

let b: New = {
  fieldName: "test",
  typoName: "error", // this causes a compile  error
}
```